Implemented Book Detail with Availability Status and Comments

SalvadorCarnelutti opened the pull request:
## Summary ##

**book-detail**: Implemented book detail to show:
- Title
- Availability Status
- Author
- Year Released
- Genre

In said view one can also see the users comments of the book, while displaying:
- Username
- User Profile Picture
- Comment

Also made connection between the book detail view and the previous library view so it can be jumped between a specific book and the whole array of book presented.

A brief description of the pull request

## Screenshots ##
![image](https://user-images.githubusercontent.com/50329976/58978107-8b6a6180-87a1-11e9-86d1-eb05ecec83fa.png)

![image](https://user-images.githubusercontent.com/50329976/58978117-91604280-87a1-11e9-8f47-39ba7ed1bd37.png)

## Known issues ##

Can't seem to shrink detail view long titles even when modifying from the Attributes Inspector.

Commit:
Implemented Book Detail with Availability Status and Comments

acid-wolox commented on WBooks/AppDelegate.swift:
```diff
@@ -28,8 +28,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         window = UIWindow(frame: UIScreen.main.bounds)
         //Desde donde empieza la aplicacion.
         window?.rootViewController = TabBarController()
+        //window?.rootViewController = UINavigationController(rootViewController: BookDetailViewController())
```
If you're not using this line in the future, remove it.

acid-wolox commented on WBooks/Views/Book Detail/BookDetailViewController.swift:
```diff
@@ -0,0 +1,86 @@
+        let data = try? Data(contentsOf: imageUrl ?? "http://wolox-training.s3.amazonaws.com/uploads/6942334-M.jpg")
+        if let imageData = data {
+            let bookImage = UIImage(data: imageData)
+            _view.imgBookCover.image = bookImage
```
Is there a reason why you split this into two lines? I think you can do:
`_view.imgBookCover.image = UIImage(data: imageData)`

acid-wolox replied:
Same as the other one, I'd swap the order of some lines and give some space:

```
        let imageUrl = URL(string: _viewModel.image)
        let data = try? Data(contentsOf: imageUrl ?? "http://wolox-training.s3.amazonaws.com/uploads/6942334-M.jpg")

        _view.lblBookTitle.text = _viewModel.title
        _view.lblStatus.text = _viewModel.status.capitalized
        _view.lblAuthor.text = _viewModel.author
        _view.lblYear.text = _viewModel.year
        _view.lblGenre.text = _viewModel.genre

        if let imageData = data {
            _view.imgBookCover.image = UIImage(data: imageData)
        }
```

SalvadorCarnelutti replied:
When you tell me to give some space, what do you mean?

acid-wolox commented on WBooks/Views/Book Detail/BookDetailViewController.swift:
```diff
@@ -0,0 +1,86 @@
+        let data = try? Data(contentsOf: imageUrl ?? "https://goo.gl/1PBWVM")
+        if let imageData = data {
+            let bookImage = UIImage(data: imageData)
+            cell.imageUser.image = bookImage
```
Is there a reason why you split this into two lines? I think you can do:
`cell.imageUser.image = UIImage(data: imageData)`

acid-wolox replied:
Also, I'd reorder and give some space to the syntax:
```
        let cell = tableView.dequeue(cell: BookDetailCell.self)!
        let commentView = _bookDetailViewModel.comments[indexPath.row]
        let imageUrl = URL(string: commentView.user.image)
        let data = try? Data(contentsOf: imageUrl ?? "https://goo.gl/1PBWVM")

        cell.lblUser.text = commentView.user.username
        cell.lblComment.text = commentView.content

        if let imageData = data {
            cell.imageUser.image = UIImage(data: imageData
        }
```

acid-wolox commented on WBooks/Views/Library/LibraryViewController.swift:
```diff
@@ -84,4 +74,33 @@ extension LibraryViewController: UITableViewDataSource, UITableViewDelegate {
+        let book = _viewModel.books[indexPath.row]
+        //********let bookDetailViewModel = BookDetailViewModel(bookVM: book)
+        //********let bookDetailViewController = BookDetailViewController(viewModel: bookDetailViewModel)
+        //********let bookDetailViewController = BookDetailViewController(viewModel: book)
```
These comments shouldn't be in the final version for a PR, as they already were committed to the branch previously and will remain there.

acid-wolox reviewed (changes requested):
Awesome work here, Salvador! There are some comments about minor things, about style, mainly.

Commit:
Made PR changes to remove unnecessary comments and superfluous lines of code.

Commit:
Updated detail view so that now it is a whole table view with the details of the book inserted as a table header

acid-wolox reviewed (approved):
Great work!
